packages/components/src/components/amount-input/amount-input.tsx

```diff
@@ -0,0 +1,151 @@
+            const input_value = e.target.value.replace(/[^\d.,]/g, '');
+            // understand the value user wants to paste:
+            const pasted_value = pasted_string
+                .replace(/[^\d.,]/g, '') // remove all characters that are not digit / point / comma
+                .replace(/[,.](?=.*[,.])/g, '') // leave only the last point / comma
+                .replace(',', '.'); // make the last point / comma separator a point
+            if (pasted_value.replace('.', '')) {
+                // if the value is a valid non-empty string, handle the two scenarios:
+                if ((value === 0 && caret_right_offset === 0) || selection_length === displayNumber(value).length) {
+                    // handle pasting when there's nothing entered before it, or it is overridden (intention: reset value):
+                    new_value = Number(
+                        pasted_value.substring(
+                            0,
+                            pasted_value.includes('.') ? max_digits + 1 : max_digits - decimal_places
+                        )
+                    );
+                } else if (input_value.replace(/[.,]/g, '').replace(/^0+/g, '').length <= max_digits) {
+                    // handle pasting when there's something entered before it and there's space for the pasted value (intention: add to value):
+                    new_value = Number(input_value.replace(/[.,]/g, '')) / Math.pow(10, decimal_places);
+                }
+            }
+        }
+        setValue(new_value);
+        setIsPasting(false);
+        onChange?.(new_value);
+    };
+
+    const inputActionHandler: React.ComponentProps<typeof Input>['onMouseDown'] &
+        React.ComponentProps<typeof Input>['onMouseUp'] &
+        React.ComponentProps<typeof Input>['onKeyDown'] = e => {
+        if (e.currentTarget.selectionStart !== null && e.currentTarget.selectionEnd !== null) {
+            setCaretRightOffset(e.currentTarget.value.length - e.currentTarget.selectionEnd);
+            setSelection({
+                selectionStart: e.currentTarget.selectionStart,
+                selectionEnd: e.currentTarget.selectionEnd,
+            });
+        }
+    };
+
+    return (
+        <div className='amount-input-wrapper'>
+            <Text size={isMobile() ? 'xxs' : 'xs'}>{label}</Text>
+            <div
+                className={classNames('amount-input-container', {
+                    'amount-input-container--error': has_error,
+                    'amount-input-container--disabled': disabled,
+                })}
+            >
+                <Input
+                    className='amount-input'
+                    disabled={disabled || focus}
+                    type='text'
+                    inputMode='numeric'
+                    value={`${displayNumber(value)} ${currency}`}
+                    // Temporary workaround to avoid jest warnings
+                    onChange={() => undefined}
```

Comment:
@sergei-deriv @mahdiyeh-deriv why is onChangeHandler not passed here? what is this workaround about? If this input is not supposed to be interactive and only needs to display a value, please change it to div. :)

Reply:
@maryia-deriv This PR is the first version of the wallets project which we want to merge it to master to prevent more gaps between this branch and master till the whole feature get ready, so this input is not connected to the functionality yet and will be fixed later on.